packages/common/src/models/Collectible.ts

```diff
@@ -32,8 +34,13 @@ export type Collectible = {
   dateLastTransferred: Nullable<string>
   externalLink: Nullable<string>
   permaLink: Nullable<string>
-  assetContractAddress: Nullable<string>
   chain: Chain
   wallet: string
   duration?: number
+
+  // ethereum nfts
+  assetContractAddress: Nullable<string>
+
+  // solana nfts
+  chainMetadata: Nullable<Metadata>
```

Comment:
could we name these something more self documenting

Reply:
hmmm, kind of hard to come up with a name here. was going to just call it `metadata`, as you can even see that its actual type from the library, but people may confuse it to the nft metadata obtained from the nft uri.

it basically represents this picture below, which is data from the metadata account, i added the `chain` prefix to show it's metadata represented on-chain.

<img width="315" alt="Screen Shot 2022-12-19 at 9 44 11 AM" src="https://user-images.githubusercontent.com/9600175/208451591-8e40121e-8095-4e83-8ee7-609106beda9b.png">

any better names come to mind?

Reply:
makes sense, how about `solanaChainMetadata` just so it's clear which chain this is from & can lose the comment?

Reply:
If we really wanted this to be correct, it could be a discriminated union on `Chain` where the solana type only exists if it's a solana chain and eth only if it's an eth chain :)